src/Microsoft.TestPlatform.Client/DesignMode/DesignModeClient.cs

```diff
@@ -287,27 +287,34 @@ private void StartTestRun(TestRunRequestPayload testRunPayload, ITestRequestMana
             Task.Run(
             delegate
             {
+                var success = false;
+                var exception = default(Exception);
                 try
                 {
                     testRequestManager.ResetOptions();
 
                     var customLauncher = skipTestHostLaunch ?
                         DesignModeTestHostLauncherFactory.GetCustomHostLauncherForTestRun(this, testRunPayload) : null;
 
-                    testRequestManager.RunTests(testRunPayload, customLauncher, new DesignModeTestEventsRegistrar(this), this.protocolConfig);
+                    success = testRequestManager.RunTests(testRunPayload, customLauncher, new DesignModeTestEventsRegistrar(this), this.protocolConfig);
                 }
                 catch (Exception ex)
                 {
                     EqtTrace.Error("DesignModeClient: Exception in StartTestRun: " + ex);
 
-                    // If there is an exception during test run request creation or some time during the process
-                    // In such cases, TestPlatform will never send a TestRunComplete event and IDE need to be sent a run complete message
-                    // We need recoverability in translationlayer-designmode scenarios
-                    var testMessagePayload = new TestMessagePayload { MessageLevel = TestMessageLevel.Error, Message = ex.ToString() };
+                    success = false;
+                    exception = ex;
+                }
+
+                // In test run failure scenario, send failure message and completion event to transaltion layer.
+                if (!success)
+                {
+                    var testMessagePayload = new TestMessagePayload { MessageLevel = TestMessageLevel.Error, Message = exception?.ToString() ?? "Exception in StartTestRun." };
```

Comment:
Suggestion for better error message in case success false comes from TestRequestManager (which today is not possible)

Reply:
Speak to pratapl, put it in resx

---
In reply to: [168675339](https://github.com/Microsoft/vstest/pull/1434#discussion_r168675339) [](ancestors = 168675339)